app/screens/home/SettingScreen.tsx

```diff
@@ -150,20 +150,15 @@ const SettingScreen = (props: any): any => {
   const actionSheetRef = createRef<any>();
   const actionSheetRefImport = createRef<any>();
   const countryId = useAppSelector(
-    (state: any) => state.selectedCountry.countryId,
+    (state: any) => state.selectedCountry?.countryId,
   );
   const allCountries = useAppSelector(
-    (state: any) => {
-     try {
-       state.selectedCountry.countries !== '' ? JSON.parse(state.selectedCountry.countries) : [];
-     } catch (error) {
-       console.error('Failed to parse countries JSON:', error);
-       return [];
-     }
-   });
+    (state: any) =>
+      state.selectedCountry.countries != '' ? JSON.parse(state.selectedCountry.countries) : [],

```

Comment:
How come selectedCountry contains multiple countries? This is against OOP concept. 
Why are we removing try/catch. How do you handle if parse method throws error?

Reply:
If we used try/catch its shown undefine error on settings screen that why we removed the try/catch.

Reply:
Then let's wrap it up with if condition to make sure state.selectedCountry is defined before accessing its property.